src/main/java/org/jitsi/srtp/SrtpCryptoContext.java

```diff
@@ -448,6 +448,46 @@ private void cryptexPostprocess(ByteArrayBuffer pkt)
         ByteArrayUtils.writeInt(pkt, headerOffset, headerTypeAndLen);
     }
 
+    /** Query whether this packet needs a zero-length header inserted.  This is used for packets that have
+     *  CSRCs but no header extensions of their own, when we are using cryptex.
+     */
+    private boolean needZeroLengthHeader(ByteArrayBuffer pkt)
+    {
+        return policy.isCryptexEnabled() && !SrtpPacketUtils.getExtensionBit(pkt) &&
+            SrtpPacketUtils.getCsrcCount(pkt) > 0;
+    }
+
+    /** A packet that has CSRCs but no header extension, when we are using cryptex.
+     * Insert a zero-length header extension so we can correctly signal that cryptex was used.
+     * Assumes needZeroLengthHeader returned true.
+     */
+    private void insertZeroLengthHeader(ByteArrayBuffer pkt)
+    {
+        int cc = SrtpPacketUtils.getCsrcCount(pkt);
+        int headerOffset = SrtpPacketUtils.FIXED_HEADER_SIZE + cc * 4;
+
+        if (pkt.getOffset() >= 4)
+        {
+            /* Move fixed header and CSRCs back. */
+            System.arraycopy(pkt.getBuffer(), pkt.getOffset(), pkt.getBuffer(), pkt.getOffset() - 4,
+                headerOffset);
+            pkt.setOffset(pkt.getOffset() - 4);
+        }
+        else
+        {
+            /* Move payload forward. */
+            if (pkt.getBuffer().length < pkt.getOffset() + pkt.getLength() + 4) {
+                /* Need more buffer. */
```

Comment:
We're encrypting a packet so we're bound to need even more for the auth tag. I don't know if it's worth optimising...